libs/forecasting/src/lib/ForecastingDashboard/ForecastingDashboard.tsx

```diff
@@ -1,19 +1,101 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-import { Text } from "@fluentui/react";
+import { Dropdown, IDropdownOption, Stack, Text } from "@fluentui/react";
 import {
   defaultModelAssessmentContext,
   ModelAssessmentContext
 } from "@responsible-ai/core-ui";
+import { localization } from "@responsible-ai/localization";
 import React from "react";
 
-export class ForecastingDashboard extends React.Component {
+import { ForecastComparison } from "./Controls/ForecastComparison";
+import { forecastingDashboardStyles } from "./ForecastingDashboard.styles";
+
+export class IForecastingDashboardProps {}
+
+export class IForecastingDashboardState {}
+
+export class ForecastingDashboard extends React.Component<
+  IForecastingDashboardProps,
+  IForecastingDashboardState
+> {
   public static contextType = ModelAssessmentContext;
   public context: React.ContextType<typeof ModelAssessmentContext> =
     defaultModelAssessmentContext;
 
   public render(): React.ReactNode {
-    return <Text>Placeholder</Text>;
+    const classNames = forecastingDashboardStyles();
+
+    if (
+      this.context === undefined ||
+      this.context.baseErrorCohort === undefined
+    ) {
+      return;
+    }
+
+    // "All data" cohort selected, so no particular time series selected yet.
+    // special case: only 1 time series in dataset, needs to be handled! TODO
+    const noCohortSelected =
+      this.context.baseErrorCohort.cohort.name ===
+      localization.ErrorAnalysis.Cohort.defaultLabel;
+
+    return (
+      <Stack
+        className={classNames.sectionStack}
+        tokens={{ childrenGap: "24px", padding: "0 0 100px 0" }}
+        id="ForecastingDashboard"
+      >
+        <Stack.Item className={classNames.topLevelDescriptionText}>
+          <Text>{localization.Forecasting.whatIfDescription}</Text>
+        </Stack.Item>
+        <Stack.Item>
+          <Dropdown
+            label={localization.Forecasting.timeSeries}
+            className={classNames.dropdown}
+            options={this.context.errorCohorts
+              .filter(
+                (cohort) =>
+                  cohort.cohort.name !==
+                  localization.ErrorAnalysis.Cohort.defaultLabel
+              )
+              .map((cohort) => {
+                return {
```

Comment:
nit: can we have this as a function?

Reply:
Certainly!

Reply:
Actually, moving this out of the JSX portion helps a ton in making it more readable. Thanks for pointing me to this unsightly code 🙂 . I don't think breaking it up further will help in that direction, but do reopen if you disagree.